docs/milestone2_walkthrough.md

```diff
@@ -0,0 +1,167 @@
+# Milestone 2 Walkthrough
+
+## Walkthrough Goals
+
+* Get a local environment running with 1 control plane cluster and 2 workload clusters
+* Define a Gateway in the control plane with a HTTPS listener, and have it instantiated in 2 workload clusters
+* Deploy an Application with a HTTPRoute to the 2 workload clusters, and have the route attached to the Gateway listener
+* Curl the Application host with DNS resolving from Route 53 and a TLS Certificate
+* Apply a Kuadrant RateLimitPolicy to the Application
+* Curl the Application host and see the rate limit being enforced
+
+## Prerequisites
+
+* [Docker Desktop](https://www.docker.com/products/docker-desktop/)
+* [AWS Account](https://aws.amazon.com/) with [Route53](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/Welcome.html) Zone permissions
+    * Choose an existing Hosted zone or create a new one for the walkthrough.
+* openssl>=3
+    * On macos a later version is available with `brew install openssl`. You'll need to update your PATH as macos provides an older version via libressl as well
+    * On fedora use `dnf install openssl`
+
+## Environment Setup
+
+Create a set of environment files with your AWS configuration.
+First, create a file called `aws-credentials.env` in the root of the project and add the following config, replacing values as necessary
+
+```bash
+AWS_ACCESS_KEY_ID=<AWS_ACCESS_KEY_ID>
+AWS_SECRET_ACCESS_KEY=<AWS_SECRET_ACCESS_KEY>
+AWS_REGION=<AWS_REGION>
+```
+
+Next, create a file called `controller-config.env` in the root of the project and add the following config, replacing values as necessary. For example, if your Hosted zone is for the domain `apps.example.com`, use that as the `ZONE_ROOT_DOMAIN`. This will mean your Application can use a hostname of `myapp.apps.example.com`
+
+```bash
+ZONE_ROOT_DOMAIN=<my.hosted.zone.domain.name>
+AWS_DNS_PUBLIC_ZONE_ID=<my.hosted.zone.id>
+```
+
+Start the local clusters and run the controller in the control plane
+
+```bash
+MCTC_WORKLOAD_CLUSTERS_COUNT=2 make local-setup
+export KUBECONFIG=./tmp/kubeconfigs/mctc-control-plane.kubeconfig
+(export $(cat ./controller-config.env | xargs) && export $(cat ./aws-credentials.env | xargs) && make build-controller install run-controller)
+```
+
+In a new terminal, create the multi cluster Gateway Class and tenant namespace
+
+```bash
+export KUBECONFIG=./tmp/kubeconfigs/mctc-control-plane.kubeconfig
+kubectl apply -f config/samples/gatewayclass.yaml
+kubectl create namespace mctc-tenant
+```
+
+Start the syncer component in both workload clusters
```

Comment:
im getting the following error when i run this
`kind load docker-image syncer:latest --name mctc-workload-1  --nodes mctc-workload-1-control-plane
Image: "" with ID "sha256:6707df96e276b056132f08658acad0f081c78158dd3db2e540dab175ed29339d" not yet present on node "mctc-workload-1-control-plane", loading...
test -s /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/controller-gen && /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/controller-gen --version | grep -q v0.10.0 || \
        GOBIN=/Users/rachellawton/Projects/multi-cluster-traffic-controller/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.10.0
/Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/controller-gen rbac:roleName=manager-role crd paths="./..." output:crd:artifacts:config=config/crd/bases
test -s /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/kustomize || { curl -Ss "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s -- 4.5.4 /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin; }
cd config/syncer && /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/kustomize edit set image syncer=syncer:latest
/Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/kustomize build config/syncer | kubectl apply -f -
Error: accumulating resources: accumulation err='accumulating resources from 'secret.yaml': evalsymlink failure on '/Users/rachellawton/Projects/multi-cluster-traffic-controller/config/syncer/secret.yaml' : lstat /Users/rachellawton/Projects/multi-cluster-traffic-controller/config/syncer/secret.yaml: no such file or directory': evalsymlink failure on '/Users/rachellawton/Projects/multi-cluster-traffic-controller/config/syncer/secret.yaml' : lstat /Users/rachellawton/Projects/multi-cluster-traffic-controller/config/syncer/secret.yaml: no such file or directory
error: no objects passed to apply
make: *** [deploy-syncer] Error 1
Error from server (NotFound): deployments.apps "sync-agent" not found
Image: "" with ID "sha256:6707df96e276b056132f08658acad0f081c78158dd3db2e540dab175ed29339d" not yet present on node "mctc-workload-2-control-plane", loading...
test -s /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/controller-gen && /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/controller-gen --version | grep -q v0.10.0 || \
        GOBIN=/Users/rachellawton/Projects/multi-cluster-traffic-controller/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.10.0
/Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/controller-gen rbac:roleName=manager-role crd paths="./..." output:crd:artifacts:config=config/crd/bases
test -s /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/kustomize || { curl -Ss "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s -- 4.5.4 /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin; }
cd config/syncer && /Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/kustomize edit set image syncer=syncer:latest
/Users/rachellawton/Projects/multi-cluster-traffic-controller/bin/kustomize build config/syncer | kubectl apply -f -
Error: accumulating resources: accumulation err='accumulating resources from 'secret.yaml': evalsymlink failure on '/Users/rachellawton/Projects/multi-cluster-traffic-controller/config/syncer/secret.yaml' : lstat /Users/rachellawton/Projects/multi-cluster-traffic-controller/config/syncer/secret.yaml: no such file or directory': evalsymlink failure on '/Users/rachellawton/Projects/multi-cluster-traffic-controller/config/syncer/secret.yaml' : lstat /Users/rachellawton/Projects/multi-cluster-traffic-controller/config/syncer/secret.yaml: no such file or directory
error: no objects passed to apply
make: *** [deploy-syncer] Error 1
Error from server (NotFound): deployments.apps "sync-agent" not found`

Reply:
and the same for the workload 2 cluster 2 its looking for secret.yaml which doesn't exist

Reply:
I've removed secret.yaml from config/syncer/kustomization.yaml after chatting with @sergioifg94 
The syncer deploys and runs OK now, but I don't see the mctc-downstream namespace or the Gateway being synced. I've left a comment asking for assistance